### ISSUE TITLE
Support reading ` .gitignore`  if present to ignore paths

### ISSUE THREAD

**Issue by NukeManDan** [becheran/mlc#77]
**Is your feature request related to a problem? Please describe.**

I don't want to duplicate the paths listed in `.gitignore` in `mlc` config. I would like the option (on by default) to use `.gitignore` as a baseline, and then respect ignoring any additional paths defined in the config.

**Describe the solution you'd like**

Ideally detecting this present in the `pwd` and/or target path to do check on and using it automatically would be great. Alternative patters:
- define the file for paths to ignore https://stackoverflow.com/a/66409885
- have an ignore file import other files https://github.com/prettier/prettier/issues/8506#issuecomment-659604909
- a way to ignore global presets https://github.com/prettier/prettier/issues/8192

**Describe alternatives you've considered**

Duplication of the ignore settings in this and other tools :upside_down_face:

**Comment by becheran** [comment]
Good idea. It would require to add and support the full gitignore syntax in mlc which is currently not the case.

**Comment by NukeManDan** [comment]
Looks like almost a drop in replacement for https://docs.rs/walkdir/latest/walkdir/ could be https://docs.rs/ignore/latest/ignore/ :thinking: might need some more CLI and config tweaks to select specifics about ignoring things. I suspect almost all users would want to default to ignoring `.gitignore` and `.ignore` files.

**Comment by becheran** [comment]
Added optinal `.gitignore` with: #94
